### PR TITLE
Integrate checks for check-gke-ingress

### PR DESCRIPTION
**Pull request by songrx1997**
/assign @swetharepakula

### CODE REVIEW

File: cmd/check-gke-ingress/app/ingress/ingress.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+			Kind:      "Ingress",
+			Namespace: ingress.Namespace,
+			Name:      ingress.Name,
+			Checks:    []*report.Check{},
+		}
+		ingressChecker := &IngressChecker{
+			client:  client,
+			ingress: &ingress,
+		}
+		ingressChecks := []ingressCheckFunc{
```
Comment (swetharepakula) [inline cmd/check-gke-ingress/app/ingress/ingress.go:56]: nit: add the check lists outside of the loop at the beginning of the function. It makes it easier to see all the checks in one shot.

File: cmd/check-gke-ingress/app/ingress/rule_test.go [hunk 2]
```diff
@@ -584,9 +627,114 @@ func TestCheckL7ILBNegAnnotation(t *testing.T) {
...
+			isL7ILB: true,
+		}
+		_, res, _ := CheckL7ILBNegAnnotation(checker)
 		if res != tc.expect {
 			t.Errorf("For test case %q, expect check result = %s, but got %s", tc.desc, tc.expect, res)
 		}
 	}
 }
+
+func TestCheckAllIngresses(t *testing.T) {
```
Comment (swetharepakula) [inline cmd/check-gke-ingress/app/ingress/rule_test.go:641]: add a comment that this just validates that all checks are run

**Review by swetharepakula — COMMENTED** [review]
a couple nits left

File: cmd/check-gke-ingress/app/ingress/rule_test.go (hunk 2 above)
Comment (songrx1997) [reply cmd/check-gke-ingress/app/ingress/rule_test.go:642]: Added, thanks!

File: cmd/check-gke-ingress/app/ingress/ingress.go (hunk 1 above)
Comment (songrx1997) [reply cmd/check-gke-ingress/app/ingress/ingress.go]: Updated, thanks!

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**